### ISSUE TITLE
Core dumped for unknown reason

### ISSUE THREAD

**Issue by Clownshift** [bio-phys/complexespp#29]
Happens on current main branch (13078a80abd54960ecd359c79fd1bd9bc075c50b).

`complexes++ --version 1` returns:

```
Copyright (c) 2018 the complexes++ development team and contributors 
 (see the file AUTHORS for the full list of names) 
 
 complexes++ is free software: you can redistribute it and/or modify 
 it under the terms of the Lesser GNU General Public License as published by 
 the Free Software Foundation, either version 3 of the License, or 
 (at your option) any later version. 
 
 complexes++ is distributed in the hope that it will be useful, 
 but WITHOUT ANY WARRANTY; without even the implied warranty of 
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the 
 GNU General Public License for more details. 
 
 You should have received a copy of the GNU General Public License 
 along with complexes++.  If not, see <https://www.gnu.org/licenses/> 


[LOG] #######################
[LOG] # PROGRAM INFORMATION #
[LOG] #######################
[LOG] name: Complexes++
[LOG] version: 0.4pre
[LOG] commit: 13078a80abd54960ecd359c79fd1bd9bc075c50b
[LOG] branch: refs/heads/main
[LOG] compilation flags:  -Wall -Wextra -Wshadow -Wnon-virtual-dtor -pedantic -Woverloaded-virtual -Wpointer-arith -Wcast-qual -Wconversion -Wold-style-cast -fopenmp -Werror  -mavx2 -march=native -mavx -march=native -msse4 -msse4.2 -march=native -msse4 -msse4.1 -march=native -mssse3 -march=native -msse3 -march=native -Wold-style-cast
[LOG] compilation libs: /tmp/paq/complexes-pp/build/complexes-pp_dynamic/ext/xdrfile-1.1.4/1.1.4/lib/libxdrfile.a;/home/paq/complexes-pp/dep/lib/libyaml-cpp.a;fmt::fmt-header-only
[LOG] 
[LOG] total runtime = 6.5925e-05s
```

With the input from the Linker tutorial (This is one of the most basic examples), I simply receive 
`Segmentation fault (core dumped)`  (see picture)
Sometimes, I also get:
```
free(): invalid pointer
Aborted (core dumped)
```
But I cannot really reproduce that...

![Screenshot from 2022-03-22 18-05-51](https://user-images.githubusercontent.com/20798138/159536768-6d3b8402-95f0-4845-b8ae-8f46fed387cf.png)
.

**Comment by Clownshift** [comment]
I tried building dynamically with and without the `debug` flag, but it does not make any difference.

**Comment by Clownshift** [comment]
Apparently, same behaviour at 1c1e6241509b98782d5acd20c440ad8ef3046188 already.

**Comment by Clownshift** [comment]
If I try the REMC tutorial, which utilizes the multidir functionality, I even get
`double free or corruption (out)` sometimes. But not always...

**Comment by Clownshift** [comment]
With the files below, you have a minimal example for testing that should theoretically work. Don't know how else to help with debugging this. 
https://syncandshare.lrz.de/getlink/fiJmvKzqpT1LUphGYN5wyF5P/test_binding.tar.xz

**Comment by berenger-eu** [comment]
Ok I am on it, what I currently get is an error of the command line module CLI11.

**Comment by berenger-eu** [comment]
Should be fixed in main, can you confirm?

**Comment by Clownshift** [comment]
Not really. It is different now though.
With 31daf7bf36d9d3e489ac919026bd176abcd1cdc9 the core dumps at `complexes++ --help`. 
I think it dumps later though, because I at least get the copyright printout. 
Btw, I also get a really weird output, if I give a false input, such as `complexes++ --version` (this is an incomplete input, because `--version` expects an argument, such as 1.

**Comment by Clownshift** [comment]
for `complexes++ --help` I get:
```
[...]
 You should have received a copy of the GNU General Public License 
 along with complexes++.  If not, see <https://www.gnu.org/licenses/> 

terminate called after throwing an instance of 'std::invalid_argument'
  what():   <-- does not exist.

Aborted (core dumped)
```

**Comment by berenger-eu** [comment]
Ok, it is my fault then for this point.

**Comment by Clownshift** [comment]
Fixed with #30 and #31
